### PR TITLE
Upgrade to Vue 2.7

### LINKED ISSUE #8078: Upgrade to Vue 2.7
This should be done before we look at Vue 3.x. 2.7 was the last 2.x version last I saw.

Need to do this after Nuxt removed and other dependency updates.

### PR DESCRIPTION
**Pull request by cnotv**
### Summary
Fixes #8078

### Technical notes summary
Followed instruction as defined in the migration guide, adding them one commit at the time to ensure passing tests.

- `eslint-plugin-vue` upgraded to 9.10.0
- Pinned existing libraries which fail on automatic upgrade
- Added linting rules to prevent the build to fail
- Corrected some linting which cannot be overruled
- Updated `yarn.lock` to ensure dependencies to work properly
- Updated test app and package configuration
- Added resolution to enforce use of Webpack 4

### CODE REVIEW

**Comment by cnotv** [comment]
CI is generated since `vue` and `vue-template-compiler` libraries upgrade.

**Comment by cnotv** [comment]
Analyzing the issue in the validate script, it has been narrowed down to the library `@types/dompurify 3.0.1` to cause linting issues with his dependency `@types/jsdom 21.1.1`

**Comment by cnotv** [comment]
Been able to reproduce issue after updating the scripts based on local requirements, such like:
- typing credentials
- adding `--registry` flag to publish
- replace publish command for libraries from `npm` to `yarn`

File: .eslintrc.js [hunk 1]
```diff
@@ -167,6 +167,15 @@ module.exports = {
...
     'vue/no-v-html':         'error',
+
+    // FIXME: Disabled rules due upgrade to Vue 2.7.x and ESLint 9.x.x1
+    'vue/multi-word-component-names':             'off',
+    'vue/no-reserved-component-names':            'off',
+    'vue/no-useless-template-attributes':         'off',
+    'vue/attribute-hyphenation':                  'off',
+    'vue/valid-next-tick':                        'off',
+    'vue/no-computed-properties-in-data':         'off',
+    'vue/no-side-effects-in-computed-properties': 'off',
```
Comment (codyrancher) [inline .eslintrc.js:178]: I was able to run `yarn lint` with this commented out. Is this still needed?

**Review by codyrancher — APPROVED** [review]
Other than my one comment I think this is fine. I clicked through cluster mangement, fleet and cluster explorer without issue. We may still want someone to take a look at harvester.

File: .eslintrc.js (hunk 1 above)
Comment (cnotv) [reply .eslintrc.js:178]: @mantis-toboggan-md guess what 😅

**Comment by cnotv** [comment]
@mantis-toboggan-md Could you please take a look to Harvester functionalities, please?

File: .eslintrc.js (hunk 1 above)
Comment (cnotv) [reply .eslintrc.js:178]: I removed the `node_module` and reinstalled everything.
With the new rules no issues, while if I comment them out I get the error for all the rules I turned off:

```
✖ 422 problems (389 errors, 33 warnings)
  0 errors and 33 warnings potentially fixable with the `--fix` option.
```

Comment (cnotv) [reply .eslintrc.js:178]: This was the issue when I tried it on push on my fork: https://github.com/cnotv/dashboard/actions/runs/4669320342/jobs/8267601781

Comment (codyrancher) [reply .eslintrc.js:178]: Alright, after deleting node_modules I see it too.

**Review by rak-phillip — APPROVED** [review]
Looks great - I did some testing with `@rancher/components`

**Comment by mantis-toboggan-md** [comment]
I imported a Harvester cluster and was not able to load the Harvester extension UI
Screen Shot 2023-04-25 at 7 24 03 PM

**Comment by nwmac** [comment]
@mantis-toboggan-md i saw that error with okta auth when disabling the auth provide and getting logged out - fixed in that PR, but that was a logout issue - worth digging into

**Comment by cnotv** [comment]
@mantis-toboggan-md have you tried to remove the `node_module` folder and reinstall everything?
I am luckily able to navigate through. Could you please confirm it?


https://user-images.githubusercontent.com/5009481/234532234-84a20405-b00c-4188-ae3d-4d628c0d86ca.mp4

**Review by mantis-toboggan-md — APPROVED** [review]
Okay yep, I cleared node_modules and I'm not seeing errors now. Sorry for the heartattack @cnotv

**Pull request merged**